api/src/services/xml_generation/submission_xml_assembler.py

```diff
@@ -0,0 +1,209 @@
+            self.application.application_attachments
+        )
+
+        request = XMLGenerationRequest(
+            application_data=app_form.application_response,
+            form_name=form_name,
+            pretty_print=False,
+            attachment_mapping=attachment_mapping,
+        )
+
+        response = self.xml_service.generate_xml(request)
+
+        if not response.success or response.xml_data is None:
+            logger.error(
+                f"Failed to generate XML for form {form_name}: {response.error_message}",
+                extra={
+                    "application_id": self.application.application_id,
+                    "form_name": form_name,
+                },
+            )
+            raise Exception(f"XML generation failed for form {form_name}: {response.error_message}")
+
+        return response.xml_data
+
+    def _assemble_xml_components(
+        self,
+        header_xml: str,
+        form_xmls: list[str],
+        footer_xml: str,
+        pretty_print: bool,
+    ) -> str:
+        """Assemble header, forms, and footer into complete XML structure."""
+        # Parse individual XML components
+        header_element = self._parse_xml_string(header_xml)
+        footer_element = self._parse_xml_string(footer_xml)
+        form_elements = [self._parse_xml_string(xml) for xml in form_xmls]
+
+        # Create root element with namespaces
+        nsmap = {
+            "header": "http://apply.grants.gov/system/Header-V1.0",
+            "footer": "http://apply.grants.gov/system/Footer-V1.0",
+            "glob": "http://apply.grants.gov/system/Global-V1.0",
+        }
+
+        root = lxml_etree.Element("GrantApplication", nsmap=nsmap)
+
+        # Add header (must strip XML declaration and use just the element)
+        root.append(header_element)
+
+        # Add Forms wrapper
+        forms_element = lxml_etree.SubElement(root, "Forms")
+        for form_element in form_elements:
+            forms_element.append(form_element)
+
+        # Add footer
+        root.append(footer_element)
+
+        # Generate final XML string
+        if pretty_print:
+            xml_bytes = lxml_etree.tostring(
+                root, encoding="utf-8", xml_declaration=True, pretty_print=True
+            )
+        else:
+            xml_bytes = lxml_etree.tostring(root, encoding="utf-8", xml_declaration=True)
+
+        return xml_bytes.decode("utf-8").strip()
+
+    def _parse_xml_string(self, xml_string: str) -> lxml_etree.Element:
+        """Parse XML string into element tree."""
+        try:
+            # Parse the XML string
+            parser = lxml_etree.XMLParser(remove_blank_text=True)
+            root = lxml_etree.fromstring(xml_string.encode("utf-8"), parser=parser)
+            return root
+        except Exception as e:
+            logger.error(f"Failed to parse XML string: {e}")
+            raise ValueError(f"Invalid XML string: {e}") from e
```

Comment:
Could this error contain the contents of the XML? We'd have to be careful about logging as these forms can contain a lot of PII.